lib/ansible/modules/system/mount.py

```diff
@@ -453,19 +453,16 @@ def is_bind_mounted(module, linux_mounts, dest, src=None, fstype=None):
     if get_platform() == 'Linux' and linux_mounts is not None:
         if src is None:
             # That's for unmounted/absent
-            for m in linux_mounts:
-                if m['dst'] == dest:
-                    is_mounted = True
+            if dest in linux_mounts:
+                is_mounted = True
         else:
-            mounted_src = None
-
-            for m in linux_mounts:
-                if m['dst'] == dest:
-                    mounted_src = m['src']
+            try:
```

Comment:
Please replace the exception by a simple key check. I would also prefer to have a simple condition instead of expression evaluation in the assignment:

```
if dest in linux_mounts:
    m = linux_mounts[dest]
elif m['src'] == src:
    is_mounted = True
```